etcd/requests.go

```diff
@@ -161,19 +161,9 @@ func (c *Client) SendRequest(rr *RawRequest) (*RawResponse, error) {
 				return
 			}
 
-			// Repeat canceling request until this thread is stopped
-			// because we have no idea about whether it succeeds.
-			for {
-				reqLock.Lock()
-				c.httpClient.Transport.(*http.Transport).CancelRequest(req)
-				reqLock.Unlock()
-
-				select {
-				case <-time.After(100 * time.Millisecond):
-				case <-cancelRoutine:
-					return
-				}
-			}
+			reqLock.Lock()
```

Comment:
also need to add a deferred recover() at the top of this gofunc... no matter what, a panic here shouldn't exit the process